### ISSUE TITLE
Restructuring of the modules source

### ISSUE THREAD

**Issue by deiga** [RyanTheAllmighty/Docker-Hub-API#26]
Hey Ryan,

I looked at the source of the package and felt that the structure of the code could be improved a bit.
Would you be open to suggestions or PR's to improve the structure? :)

**Comment by RyanTheAllmighty** [comment]
Yeah this was written almost 4 years ago now, it's terrible in it's current form.

Very happy to accept PR's, since I don't use this module myself as actively anymore, I haven't had a chance to spend some time to improve it
